### DISCUSSION TITLE
Customize 403 page

### DISCUSSION THREAD

**Discussion by dor1202** [OrchardCMS/OrchardCore#12672] [Q&A]
Hi,
I want to customize the default pages, I managed to style the login page by creating a theme and overriding the page called `Login.cshtml`  in the `OrchardCore.Users` folder.
I failed to style the error pages, page 403 in particular.
I'm adding a screenshot of the page with the default style and URL, any ideas how to change the style to match the theme?
![Screenshot 2022-10-20 130209](https://user-images.githubusercontent.com/71217326/196920657-edddf647-d310-4249-8ba2-8f408371742b.png)

**Comment by hishamco** [comment]
IMHO you can change or override the `OC.Diagnostics` pages

**Reply by dor1202** [reply]
Thank you for the quick response, I've added the OrchardCore.Diagnostics folder, I'm adding a screenshot of the file tree, but it still doesn't seem to work. Do you have any recommendations?
![Screenshot 2022-10-23 094359](https://user-images.githubusercontent.com/71217326/197378457-339cfb39-842e-4aca-89b6-63e7634209e1.png)

**Reply by hishamco** [reply]
We need to check, but I presume overriding the view should work

**Reply by dor1202** [reply]
@hishamco Ok, please let me know when you have time to check it 👍

**Reply by hishamco** [reply]
I'm quite busy with other things, hopefully if I can have a look at the weekend, meanwhile you might provide a minimal repo for reproduce the issue, so anyone else might help

**Reply by ChestnutYYY** [reply]
!!! I had the same problem recently, looking forward to your fix

**Reply by dor1202** [reply]
@hishamco Unfutenatly I'm pretty busy in the near weeks, I will try to create a repo for the issue as soon as I can!
@rllxxx Glad to see more people needing the fix!

**Reply by ChestnutYYY** [reply]
thanks you so much @jtkech ， the method you provided is ok, @dor1202 , you can try it agin

**Comment by jtkech** [comment]
Look at the `OC.Diagnostics` module, under its `Views/Diagnostics` folder to see all the views you can override in a given theme, in your case you need to override `Forbidden.cshtml`.

Then because it is not a shape view but a pure mvc view (related to the `DiagnosticsController`), in your theme place your view override under the following theme subfolder.

    Views/OrchardCore.Diagnostics/Diagnostics/Forbidden.cshtml

The above path uses the module and controller name to avoid conflicts.

But to override the Forbidden view you can just use the following theme subfolder.

    Views/Shared/Forbidden.cshtml

**Reply by dor1202** [reply]
Hi @jtkech , thank you for the comment. It doesn't seem to work for me, adding screenshots for clarity.
I attempted to move the `Forbidden.cshtml` from 
`Views/OrchardCore.Diagnostics/Diagnostics/Forbidden.cshtml`
To  `Views/Shared/Forbidden.cshtml` but still no change in the html page.
![Screenshot 2022-10-30 105930](https://user-images.githubusercontent.com/71217326/198870590-ec0fb4b7-0f8c-4927-91b0-6374dac87a1a.png)
![Screenshot 2022-10-30 105953](https://user-images.githubusercontent.com/71217326/198870594-1b46feaf-6cfb-4081-98f2-7238f7940994.png)
![Screenshot 2022-10-30 110018](https://user-images.githubusercontent.com/71217326/198870595-f255b57c-5bee-477e-a41a-3fcabbdec99d.png)

**Comment by jtkech** [comment]
Hmm, is your theme enabled and selected as the current theme?
Are your other overridden views working?

Try to rebuild your theme project and your application.
See in your theme project file if some extra `ItemGroup` was added, if so remove them.

**Reply by dor1202** [reply]
@jtkech The theme is enabled and selected as a default theme.
Both the `OrchardCore.Users` and `OrchardCore.OpenId` folders are working fine and overriding the pages. For example, the login and logout pages both use the custom layout.
I've tried rebuilding the theme project and the solution but no `ItemGroup` tags were added.
I'm adding the `ItemGroup` in the theme, maybe I missed something there.
![Screenshot 2022-11-02 143941](https://user-images.githubusercontent.com/71217326/199491818-6e56bed7-615c-4ab3-b5f2-77787e049767.png)

**Reply by jtkech** [reply]
Okay, that's weird. So I thought that maybe there is a difference when referencing OrchardCore as packages, but just tried it in this context and it works ;) Are other views working, for example `BadRequest.cshtml`?

Hmm, can you share a minimal project so that I can repro and test it? In the meantime can you tried by putting the view under `Views/OrchardCore.Diagnostics/Diagnostics/Forbidden.cshtml`. <= Just saw that you did it.

**Reply by dor1202** [reply]
All of the pages in `OrchardCore.Diagnostics` are not overridable from my tries:

- BadRequest.cshtml
- Error.cshtml
- Forbidden.cshtml
- NotFound.cshtml
- Unauthorized.cshtml

As I wrote to @hishamco, I will try to create a minimal project repo and tag both of you to see as soon as I can!

> In the meantime can you tried by putting the view under `Views/OrchardCore.Diagnostics/Diagnostics/Forbidden.cshtml`. <= Just saw that you did it.

Had no luck with `Views/OrchardCore.Diagnostics/Diagnostics/Forbidden.cshtml` approach.

**Reply by jtkech** [reply]
Okay cool, let me know
